src/common/TimeAgo/TimeAgo.js

```diff
@@ -0,0 +1,75 @@
+const SECS = 1000;
+const MINS = 60 * SECS;
+
+import { format, formatDistanceToNow } from "date-fns";
+import { useCallback, useEffect, useMemo, useRef, useState } from "react";
+
+// to_time, [step]
+const STAGES = [
+  [30 * SECS, 30 * SECS],
+  [44 * MINS + 30 * SECS, 1 * MINS],
+  [Number.MAX_SAFE_INTEGER, 30 * MINS],
+];
+
+function * scheduleGen(value = 0) {

```

Comment:
comments, please. and typescript to understand resulting values

Reply:
and I think it would be mush simpler to just create a calculation function like `getNextTick(timePassed)`